### PR TITLE
Move e2e tests in JIT under tests dir

### PR DESCRIPTION
**Pull request by mathetake**
This PR generalizes e2e tests in `jit/engine_test.go`and moves them
`tests/adhoc` directory to run these tests against interpreter as well.

Notably this removes all the `testdata` directory from `wasm/*`
and helps isolates the usage of raw binary or texts into `examples`,
`bench` and `tests` directory.

During the course of generalization, I found the inconsistency on the 
panic handling between JIT and Interpreter which seems a bug of latter
so I fixed it.

### CODE REVIEW

File: examples/trap_test.go [hunk 1]
```diff
@@ -1,41 +0,0 @@
-package examples
```
Comment (mathetake) [inline examples/trap_test.go:1]: this was definitely not an `example` but an e2e test, and is a dup of the e2e test used in jit previously. So I deleted it

File: examples/README.md [hunk 2]
```diff
@@ -1,4 +0,0 @@
-
-This is where we put e2e tests of virtual machine. Please place examples so that `examples/foo_test.go` is testing
```
Comment (mathetake) [inline examples/README.md:2]: this is a remnants from my hobby project day and confusing so deleted for now

File: wasm/interpreter/interpreter.go [hunk 3]
```diff
@@ -427,34 +427,46 @@ func (it *interpreter) lowerIROps(f *wasm.FunctionInstance,
...
+
+	// shouldRecover is true when a panic at the origin of callstack should be recovered
+	//
+	// If this is the recursive call into Wasm (prevFrameLen != 0), we do not recover, and delegate the
+	// recovery to the first interpreter.Call.
+	//
+	// For example, given the call stack:
+	//	 "original host function" --(interpreter.Call)--> Wasm func A --> Host func --(interpreter.Call)--> Wasm function B,
+	// if the top Wasm function panics, we go back to the "original host function".
+	shouldRecover := prevFrameLen == 0
```
Comment (mathetake) [inline wasm/interpreter/interpreter.go:439]: that was the inconsistency I found

File: tests/README.md [hunk 4]
```diff
@@ -1,3 +1,4 @@
 This directory contains tests which use multiple packages. For example:
+- `adhoc` contains variety of e2e tests, mainly to ensure the consistency in the behavior between engines.
```
Comment (codefromthecrypt) [inline tests/README.md:2]: maybe call the directory engine instead, then say the tests inside are ad-hoc?

**Review by codefromthecrypt — APPROVED** [review]
fine either way, but I think the directory may be better "engine"

**Pull request merged**